apps/wallet-connect/src/hooks/useWalletConnectV2.tsx

```diff
@@ -0,0 +1,307 @@
+import { useState, useCallback, useEffect, useMemo } from 'react'
+import SignClient from '@walletconnect/sign-client'
+import { SignClientTypes, SessionTypes, CoreTypes } from '@walletconnect/types'
+import { SafeAppProvider } from '@gnosis.pm/safe-apps-provider'
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk'
+import { ChainInfo } from '@gnosis.pm/safe-apps-sdk'
+import { ethers } from 'ethers'
+
+import {
+  NEW_SESSION_ACTION,
+  TRANSACTION_CONFIRMED_ACTION,
+  WalletConnectVersion,
+  WALLET_CONNECT_VERSION_2,
+} from '../utils/analytics'
+
+const { REACT_APP_WALLETCONNECT_PROJECT_ID, NODE_ENV } = process.env
+
+const isProduction = NODE_ENV === 'production'
```

Comment:
nit: maybe it's worth exporting constants to a separate file like `constants.ts` to centralize access to environment variables

Reply:
I'd also rename `REACT_APP_WALLETCONNECT_PROJECT_ID` to `REACT_APP_WALLETCONNECT_V2_PROJECT_ID` cos it's only relevant for v2

Reply:
done! I created a new constant `WALLETCONNECT_V2_PROJECT_ID` in the `constants.ts` file.